lib/web_ui/lib/src/engine/window.dart

```diff
@@ -153,7 +153,17 @@ class EngineWindow extends ui.Window {
 
   /// Handles the browser history integration to allow users to use the back
   /// button, etc.
-  final BrowserHistory _browserHistory = BrowserHistory();
+  BrowserHistory _browserHistory = SingleEntryBrowserHistory();
+
+  Future<void> _updateBrowserHistory({bool useRouter = false}) async {
+    if (useRouter == _browserHistory is MultiEntriesBrowserHistory) {
+      return;
+    }
+    final LocationStrategy strategy = _browserHistory.locationStrategy!;
```

Comment:
We have to always handle the case of `locationStrategy` being null. There are apps that choose to disable all browser history/url changes by setting `locationStrategy` to null.